### ISSUE TITLE
Print list of exposed binaries after `nix-env -i`

### ISSUE THREAD

**Issue by danbst** [NixOS/nix#1049]
I propose to print all exposed to PATH binaries after successful `nix-env -i`.

Here is a list of packages where I spent time with `grep` and `find` to find the binary name:
- `extremetuxracer`, binary `etr`
- `tortoisehg`, binary `thg`
- `libreoffice`, binary `soffice`

Two of above have appropriate symlinks, but this is ad-hoc solution (we shouldn't link `mercurial` to `hg`).

Partly inspired by https://github.com/NixOS/nix/issues/1010#issuecomment-239255867

**Comment by edolstra** [comment]
That could print hundreds of binaries...

It's probably better to have a command to query the contents of an installed package.

**Comment by danbst** [comment]
I'm OK with hundreds of binaries. Once upon a time, I've got this:
```
The program ‘lspci’ is currently not installed. It is provided by
several packages. You can install it by typing one of the following:
  nix-env -iA nixos.busybox
  nix-env -iA nixos.pciutils
```

So I installed `busybox` to get `lspci`. Unfortunately I completely missed that it installed hundreds of binaries, some of which didn't work (in particular, `man`). It took time to figure out who was the problem.

But now I think, that this should be a plugin to core Nix, and issue doesn't belong to this repo.
